python/ray/autoscaler/_private/monitor.py

```diff
@@ -192,12 +271,16 @@ def update_load_metrics(self):
                                  and self.autoscaler.config["provider"].get(
                                      "use_node_id_as_ip", False))
             if use_node_id_as_ip:
-                ip = str(int(total_resources.get("NODE_ID_AS_RESOURCE", 0)))
+                ip = resource_message.node_id.hex()
             else:
                 ip = resource_message.node_manager_address
-            self.load_metrics.update(
-                ip, total_resources, available_resources, resource_load,
-                waiting_bundles, infeasible_bundles, pending_placement_groups)
+            self.load_metrics.update(ip, total_resources, available_resources,
+                                     resource_load, waiting_bundles,
+                                     infeasible_bundles,
+                                     pending_placement_groups, cluster_full)
+        if self.readonly_config:
+            self.readonly_config["available_node_types"].update(
+                mirror_node_types)
```

Comment:
Ok, maybe missing resource messages aren't a problem since we accumulate everything in this dict.

What if a node is removed from the Ray cluster? Do we have logic to get it out of the readonly_config?

Reply:
We don't--- but the expectation is these clusters are fixed size.

Reply:
I guess it's P2 to deal with situations in which nodes crash or are removed by users. Worst case, messages are not as useful.